### DISCUSSION TITLE
Typing question

### DISCUSSION THREAD

**Discussion by 9alecj** [AllocZero/EfficientDynamoDb#249] [Q&A] [answered]
Hello! I have been using the AWS dynamo library and am working on switching to this library as it seems much more performant. However, I am having an issue with my existing data in dynamo, I am getting this error during a scan operation:

`System.InvalidOperationException: Attribute contains 'StringSet' value instead of 'List'.`

Any ideas how I might fix such an error?

**Comment by firenero** [comment, marked as answer]
Hi @9alecj, this error happens when List-based converter is used on a field that is stored in DynamoDB as `StringSet`. It's hard to pinpoint an exact location without seeing your code but I can suggest looking at these cases:
- During migration you've changed `HashSet<T>` to array-based type such as `List<T>` or `T[]`, etc. So under the hood the list based converter is used now.
- You had a custom converter applied to some field that was serializing value as a `StringSet`. And during migration you didn't apply the converter or migrated converter uses `List` as an underlying DynamoDB type.
- Wild guess: you used [SdkEnumHashSetConverter](https://github.com/AllocZero/EfficientDynamoDb/blob/3cccd93147f5a4ca78eac43ddef24e0bb614cec4/src/EfficientDynamoDb/Converters/Collections/SdkEnumHashSetConverter.cs#L8) on `HashSet<TEnum>` that was in fact serialized as `StringSet`. 

If nothing of these helps, I can help you more if you provide the code for your entity class. Feel free to anonymize property names, etc.

**Reply by 9alecj** [reply]
Hello, thanks for the informative and helpful response!

I just wanted to explain some background and what fixed it for me in case anyone else runs into the same issue. We have a new API created in .net 6 about 9 months ago. For this project we used the default AWS dynamo library and used the entity and table annotations that come with that. The table in question had a second layer object with a string list like so: 


```
public class ParentEntity
{
    public string Id {get;set;}
    public ChildEntity Child {get;set}
}

public class ChildEntity
{
public string Name {get;set;}
public List<string> Items {get;set;}
}
```

For some reason the Items property here was saved in dynamo as a StringSet though in code it has always been described as a List<string>. This seems to cause in incompatibility in the EfficientDynamoDb. Luckily it seems like maybe using a HashSet is a better thing to do here anyway so I swapped out List for HashSet and it seems to work now. Also for what its worth it is quite a bit faster than the default library exactly as described, thanks!
